bundles/org.openhab.binding.siemensrds/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,189 @@
+				<label>Polling Interval</label>
+				<description>Time (seconds) between polling requests to the cloud server</description>
+				<default>60</default>
+			</parameter>
+
+			<parameter name="apiKey" type="text" required="true">
+				<label>Climatix IC API Key</label>
+				<description>The key needed to access the Siemens Climatix IC cloud server</description>
+			</parameter>
+ 
+		</config-description>
+
+	</bridge-type>
+
+	<thing-type id="rds">
+
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="climatixic"/>
+		</supported-bridge-type-refs>
+
+		<label>RDS Thermostat</label>
+		<description>Siemens RDS Smart Thermostat</description>
+
+		<channels>
+			<channel id="roomTemperature" typeId="temperature">
+		   		<label>Room Temperature</label>
+		   		<description>Actual room temperature</description>
+			</channel>
+
+			<channel id="setTemperature" typeId="setTemperature">
+		   		<label>Target Temperature</label>
+   				<description>Target temperature setting for the room</description>
+			</channel>
+
+			<channel id="heatingOffCooling" typeId="heatingOffCooling">
+		   		<label>Heating / Cooling</label>
+   				<description>Heating / Cooling state </description>
+			</channel>
+
+			<channel id="roomHumidity" typeId="humidity">
+		   		<label>Room Humidity</label>
+		   		<description>Actual room humidity</description>
+			</channel>
+
+			<channel id="roomAirQuality" typeId="airQuality">
+		   		<label>Room Air Quality</label>
+		   		<description>Actual room air quality</description>
+			</channel>
+
+			<channel id="outsideTemperature" typeId="temperature">		   		
+		   		<label>Outside Temperature</label>
+		   		<description>Actual outside temperature</description>
+			</channel>
+
+			<channel id="greenLeaf" typeId="greenLeaf">
+		   		<label>Green Leaf Mode</label>
+   				<description>Energy saving level</description>
+			</channel>
+
+			<channel id="statProgramMode" typeId="statProgramMode">
+				<label>T&apos;stat Program (Manual::Auto)</label>
+		   		<description>The program mode of the thermostat (user can override)</description>
+			</channel>
+
+			<channel id="statOccupancyState" typeId="statOccupancyState">
+		   		<label>Occupancy State (Absent::Present)</label>
+   				<description>The occupancy mode of the thermostat (user can override)</description>
+			</channel>
+
+			<channel id="dhwProgramMode" typeId="dhwProgramMode">
+				<label>DHW Program (Manual::Auto)</label>
+		   		<description>The program mode of the domestic water heating (user can override)</description>
+			</channel>
+			
+			<channel id="dhwOffOnState" typeId="dhwOffOnState">
+				<label>DHW Switch State (Off::On)</label>
+		   		<description>The state of the domestic water heating (user can override)</description>
+			</channel>
+
+		</channels>
+
+		<properties>
+			<property name="vendor">Siemens</property>
+			<property name="modelId">RDS</property>
+			<property name="climatixIcPlantId"></property>
```

Comment:
This must not be a property, but it must be a configuration parameter.

Reply:
Ok.

Reply:
I tried to implement this as you requested, but there is a problem that prevents it...

The PlantId (or climatixIcPlantId as it is here incorrectly called) is a unique identifier of a thermostat, which is discovered by the Discovery Service and is currently written as Property of the thermostat when it is put as a not-yet-instantiated Thing into the PaperUI Inbox.

You are logically correct that the PlantId should really be a Configuration Parameter but (as far as I can tell) there is no way for the Discovery Service to write a Configuration Parameter for a not-yet-instantiated Thing in the Inbox. I think Configuration Parameter s can only be written when the Thing is finally initiated by the user.

Reply:
> there is no way for the Discovery Service to write a Configuration Parameter

There is! You are correct that DiscoveryResults only have properties and that you have to write the PlatId as a property into the DiscoveryResult. BUT as soon as the user approves the Inbox entry, the framework checks, which properties of the DiscoveryResult have to be created as Thing configuration parameters and which as Thing properties - simply by checking whether there is a config parameter with the given id.

Reply:
Clever!

Reply:
Ok.

Reply:
Please review if it is Ok.